### PR TITLE
chore: update grpc runtime library

### PR DESCRIPTION
**Pull request by leoluz**
Signed-off-by: Leonardo Luz Almeida <leonardo_almeida@intuit.com>

# TL;DR

This PR updates the gRPC runtime library (to the current latest version v1.45.0) used in ArgoCD. 

# Analysis

In ArgoCD we have a mix of `proto3` and `proto2` used for gRPC messages:
- `proto2` is used by the Application gRPC APIs and uses gogo-protobuf annotations.
- `proto3` is used for all other operations provided by the API server.

With this scenario, updating the grpc runtime library works for most of UI/CLI operations but fail in Application ones with the error:
```
09:13:53  api-server | panic: protobuf tag not enough fields in Empty.state: 
09:13:53  api-server | goroutine 403 [running]:
09:13:53  api-server | github.com/gogo/protobuf/proto.(*unmarshalInfo).computeUnmarshalInfo(0xc000fa2500)
09:13:53  api-server | 	/Users/lalmeida1/dev/go/src/github.com/argoproj/argo-cd/vendor/github.com/gogo/protobuf/proto/table_unmarshal.go:341 +0x138a
```
Error explanation details [can be found here](https://github.com/jaegertracing/jaeger/pull/2857#issuecomment-791704993). As this is coming from the `gogo` marshaling code, one other possibility would be to remove `gogo` plugin from the project and use the standard provided by google.protobuf. Unfortunately this is still not possible because of Kubernetes proto files incompatibility issues [described here](https://github.com/argoproj/argo-rollouts/issues/1091#issuecomment-821448401).

# Solution

With this scenario, the only possibility to update gRPC runtime library in ArgoCD is by removing all `gogo` annotations from the `application.proto` so we don't need to rely on generated Marshallers code and avoid the incompatibility issue. However we still need to use `gogo` plugin for go stubs generation to maintain the compatibility with Kubernetes protos.

This PR is an attempt to implement the solution above.

Relates to https://github.com/argoproj/argo-cd/issues/4972

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Review by alexmt — APPROVED** [review]
LGTM.

Code changes look good. Tested CLI locally as well and could not find any bugs. Lets merge it !

File: pkg/apiclient/application/application.pb.go [hunk 1]
```diff
@@ -44,13 +43,13 @@ type ApplicationQuery struct {
 	// forces application reconciliation if set to true
 	Refresh *string `protobuf:"bytes,2,opt,name=refresh" json:"refresh,omitempty"`
 	// the project names to restrict returned list applications
-	Projects []string `protobuf:"bytes,3,rep,name=project" json:"project,omitempty"`
+	Projects []string `protobuf:"bytes,3,rep,name=projects" json:"projects,omitempty"`
```
Comment (victorboissiere) [inline pkg/apiclient/application/application.pb.go:46]: FYI, this change deleted all of our apps since we used `project=XX`. The filter on project was not there and resulted in all our apps being listed and deleted. This was not in the changelog.

Comment (crenshaw-dev) [reply pkg/apiclient/application/application.pb.go:46]: @victorboissiere would you mind opening an issue requesting that the upgrade guide be changed to include a note about this?

We probably also need to establish a stricter policy about API backwards-compatibility.

Comment (jannfis) [reply pkg/apiclient/application/application.pb.go:46]: > We probably also need to establish a stricter policy about API backwards-compatibility.

I think we already promise backwards compatibility for the API. And I think this one just slipped through the cracks.

Comment (leoluz) [reply pkg/apiclient/application/application.pb.go:46]: This file is auto-generated. The breaking change [originated here](https://github.com/argoproj/argo-cd/pull/8929/files#diff-df6e82b174b5eaec332889c540b09c2fa66b2eb2bcd1b40545cfd1d2d1800bc2R23). This was probably a misunderstanding about how the `gogoproto.customname` worked. Unfortunately this wasn't caught by our automated tests on that occasion. I agree with @jannfis that a stricter policy isn't the answer to this incident. Our API is exposed at `/api/v1` and that shouldn't break. What we could do to avoid this type of problem in the future is improving our automated tests. Probably having some sort of contract testing would be beneficial.

Comment (leoluz) [reply pkg/apiclient/application/application.pb.go:46]: @victorboissiere I am sorry about the incident that this change caused. To avoid this type of issue to happen again in the future, I created the following proposal: https://github.com/argoproj/argo-cd/issues/12589

Comment (crenshaw-dev) [reply pkg/apiclient/application/application.pb.go:46]: @victorboissiere what was your client? I'm investigating whether this is a problem with the CLI.

Comment (victorboissiere) [reply pkg/apiclient/application/application.pb.go:46]: Thanks for the issue! it was not the CLI, we’ve built an internal tool that cleanup from time to time some project applications. The filter stopped working and we’ve lost ~8000 pods. 
Thanks for the effort to avoid this in the future 🙏

Comment (crenshaw-dev) [reply pkg/apiclient/application/application.pb.go:46]: Sweet, thanks! PR to fix the issue: https://github.com/argoproj/argo-cd/pull/12594

I had written up some warnings about the CLI, but after investigation, I don't think the CLI is impacted. Only the JSON API.
